tests/src/analysis/testqgsrastercalculator.cpp

```diff
@@ -876,6 +881,215 @@ void TestQgsRasterCalculator::testStatistics()
 
 }
 
+void TestQgsRasterCalculator::testFunctTypeFirstCase()
+{
+  // Test with one raster as condition and numbers as first and second option
+  //if (raster1 > 5.0, 100.0, 10.0)
+  QVector < QgsRasterCalcNode * > args;
+
+  //raster
+  QgsRasterBlock m1( Qgis::DataType::Float32, 2, 3 );
+  m1.setNoDataValue( -1.0 );
+  m1.setValue( 0, 0, 15.0 );
+  m1.setValue( 0, 1, 1.5 );
+  m1.setValue( 1, 0, 13.0 );
+  m1.setValue( 1, 1, -1.0 ); //nodata
+  m1.setValue( 2, 0, 11.0 );
+  m1.setValue( 2, 1, 1.0 );
+  QMap<QString, QgsRasterBlock *> rasterData;
+  rasterData.insert( QStringLiteral( "raster1" ), &m1 );
+  args.append( new QgsRasterCalcNode( QgsRasterCalcNode::opGT, new QgsRasterCalcNode( QStringLiteral( "raster1" ) ), new QgsRasterCalcNode( 5.0 ) ) );
+
+  // append the two options
+  args.append( new QgsRasterCalcNode( 100.0 ) );
+  args.append( new QgsRasterCalcNode( 10.0 ) );
+
+  //name of the function
+  QString name( "if" );
+
+  QgsRasterCalcNode node( name, args );
```

Comment:
now that parsing of "if" condition works, it will be better to use the parser instead of manually creating the expression - it will be also much easier to read/understand the test(s)